### PR TITLE
Clean sneak peek

### PR DESCRIPTION
**Pull request by jlikhuva**
@stvstnfrd I've opened this 'PR' to enable faster communication/collaboration

### CODE REVIEW

File: common/djangoapps/student/views.py [hunk 1]
```diff
@@ -471,10 +468,7 @@ def register_user(request, extra_context=None):
     if settings.FEATURES.get('USE_CME_REGISTRATION'):
         return cme_register_user(request, extra_context=extra_context)
 
-    # Determine the URL to redirect to following login:
     redirect_to = get_next_url_for_login_page(request)
-    if UserProfile.has_registered(request.user):
```
Comment (stvstnfrd) [inline common/djangoapps/student/views.py]: Change to `user.is_authenticated()`

File: common/djangoapps/student/views.py [hunk 2]
```diff
@@ -572,10 +566,6 @@ def is_course_blocked(request, redeemed_registration_codes, course_key):
 @ensure_csrf_cookie
 def dashboard(request):
     user = request.user
-
-    if not UserProfile.has_registered(user):
```
Comment (stvstnfrd) [inline common/djangoapps/student/views.py]: Yes, remove this entire block.

File: common/djangoapps/student/views.py [hunk 3]
```diff
@@ -1053,11 +1009,6 @@ def change_enrollment(request, check_access=True):
     """
     # Get the user
     user = request.user
-
-    # Ensure the user is authenticated
-    if not UserProfile.has_registered(user):
```
Comment (stvstnfrd) [inline common/djangoapps/student/views.py]: Change to `user.is_authenticated()`

File: cms/djangoapps/sneakpeek/middleware.py [hunk 4]
```diff
@@ -1,26 +0,0 @@
-"""
-Middleware class handling sneakpeek in cms, which should never be allowed
-"""
-from student.models import UserProfile
-from django.contrib.auth import logout
-from django.shortcuts import redirect
-
-
-class SneakPeekLogoutMiddleware(object):
```
Comment (stvstnfrd) [inline cms/djangoapps/sneakpeek/middleware.py:9]: We'll want to reimplement some form of this middleware for CMS.

```python
if request.user.is_lazy_user():
    logout(request)
    return redirect(request.get_full_path())
return None
```

File: cms/urls.py [hunk 5]
```diff
@@ -224,3 +224,7 @@
     url(r'^404$', handler404),
     url(r'^500$', handler500),
 )
+
+urlpatterns += (
```
Comment (stvstnfrd) [inline cms/urls.py]: Actually, since we're not allowing sneakpeek access to CMS, we can remove this `urlpattern`.

File: common/djangoapps/external_auth/tests/test_ssl.py [hunk 6]
```diff
@@ -96,25 +96,6 @@ def test_ssl_login_with_signup_lms(self):
         with self.assertRaises(User.DoesNotExist):
             User.objects.get(email=self.USER_EMAIL)
 
-    @unittest.skipUnless(settings.ROOT_URLCONF == 'cms.urls', 'Test only valid in cms')
-    def test_ssl_login_with_signup_cms(self):
```
Comment (stvstnfrd) [inline common/djangoapps/external_auth/tests/test_ssl.py]: I don't think we want to remove this;
it appears unrelated to sneakpeek.

File: common/djangoapps/student/migrations/0001_initial.py [hunk 7]
```diff
@@ -230,7 +230,6 @@ class Migration(migrations.Migration):
                 ('bio', models.CharField(max_length=3000, null=True, blank=True)),
                 ('profile_image_uploaded_at', models.DateTimeField(null=True)),
                 ('user', models.OneToOneField(related_name='profile', to=settings.AUTH_USER_MODEL)),
-                ('nonregistered', models.BooleanField(default=False)),
```
Comment (stvstnfrd) [inline common/djangoapps/student/migrations/0001_initial.py]: Instead of just removing this, I think we may need to instead add a migration to remove the column.

File: common/djangoapps/student/models.py [hunk 8]
```diff
@@ -916,13 +890,7 @@ def users_enrolled_in(self, course_id):
...
             courseenrollment__is_active=True,
         )
-        anonymous_user_ids = [
-            user.id
-            for user in course_users
-            if not UserProfile.has_registered(user)
-        ]
-        enrolled_course_users = course_users.exclude(id__in=anonymous_user_ids)
-        return enrolled_course_users
+        return course_users
```
Comment (stvstnfrd) [inline common/djangoapps/student/models.py]: We probably still want to filter out those users here;
we'll just want to check user.is_lazy_user` instead.

File: common/djangoapps/student/models.py [hunk 9]
```diff
@@ -1070,7 +1038,6 @@ def num_enrolled_in(cls, course_id):
         enrollment_number = CourseEnrollment.objects.filter(
             course_id=course_id,
             is_active=1,
-            user__profile__nonregistered=0,
```
Comment (stvstnfrd) [inline common/djangoapps/student/models.py]: Similarly, we probably still need some form of filter here.

File: common/djangoapps/student/views.py [hunk 10]
```diff
@@ -433,8 +432,6 @@ def signin_user(request):
     external_auth_response = external_auth_login(request)
     if external_auth_response is not None:
         return external_auth_response
-    if UserProfile.has_registered(request.user):
```
Comment (stvstnfrd) [inline common/djangoapps/student/views.py]: Or should this be `user.is_lazy_user`?

File: common/djangoapps/util/tests/test_submit_feedback.py [hunk 11]
```diff
@@ -124,117 +124,6 @@ def _assert_datadog_called(self, datadog_mock, with_tags):
         ]
         self.assertEqual(datadog_mock.mock_calls, expected_datadog_calls)
 
-    def test_bad_request_anon_user_no_name(self, zendesk_mock_class, datadog_mock):
```
Comment (stvstnfrd) [inline common/djangoapps/util/tests/test_submit_feedback.py]: I don't think we actually want to remove these.

File: lms/templates/navigation.html [hunk 12]
```diff
@@ -195,7 +195,7 @@ <h2 class="course-header"><span class="provider">${course.display_org_with_defau
 <![endif]-->
 % endif
 
-%if not UserProfile.has_registered(user):
+%if not user.is_authenticated():
```
Comment (stvstnfrd) [inline lms/templates/navigation.html]: The forgot password modal should only be displayed to non-lazy users.

File: common/djangoapps/util/views.py [hunk 13]
```diff
@@ -318,18 +316,6 @@ def build_error_response(status_code, field, err_msg):
         [(tag, request.POST[tag]) for tag in ["issue_type", "course_id"] if tag in request.POST]
     )
 
-    if UserProfile.has_registered(request.user):
```
Comment (stvstnfrd) [inline common/djangoapps/util/views.py:321]: I think we want this logic, just maybe need to change the `has_registered` call.

File: common/lib/xmodule/xmodule/tabs.py [hunk 14]
```diff
@@ -58,8 +58,6 @@ class CourseTab(object):
     # If there is a single view associated with this tab, this is the name of it
     view_name = None
 
-    # True if this tab can be displaed to sneak peek users
-    is_visible_to_sneak_peek = False
```
Comment (stvstnfrd) [inline common/lib/xmodule/xmodule/tabs.py]: I think we want to keep this.

File: lms/djangoapps/courseware/access.py [hunk 15]
```diff
@@ -439,49 +435,6 @@ def check_for_staff():
     return _dispatch(checkers, action, user, descriptor)
 
 
-NONREGISTERED_CATEGORY_WHITELIST = [
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/access.py]: I think we want this,
but maybe we can move it to `openedx/stanford/lms/envs/common.py`

File: lms/djangoapps/courseware/access.py [hunk 16]
```diff
@@ -574,10 +527,6 @@ def can_load():
         students to see modules.  If not, views should check the course, so we
         don't have to hit the enrollments table on every module load.
         """
-        if user.is_authenticated():
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/access.py]: I think we want some form of this check.

File: lms/djangoapps/courseware/migrations/0002_add_coursepreference.py [hunk 17]
```diff
@@ -1,28 +0,0 @@
-# -*- coding: utf-8 -*-
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/migrations/0002_add_coursepreference.py]: I think we want to keep this, but we'll remove the table via another database migration.

**Review by stvstnfrd — CHANGES_REQUESTED** [review]
work-in-progress

File: common/djangoapps/student/migrations/0001_initial.py (hunk 7 above)
Comment (stvstnfrd) [reply common/djangoapps/student/migrations/0001_initial.py]: Restore this line, then once you've removed `nonregistered` from the model file, run `manage.py makemigrations`

File: lms/djangoapps/courseware/models.py [hunk 18]
```diff
@@ -295,35 +295,6 @@ def __unicode__(self):
         return "[OCGLog] %s: %s" % (self.course_id.to_deprecated_string(), self.created)  # pylint: disable=no-member
 
 
-class CoursePreference(models.Model):
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/models.py]: Note to self: TODO: resume from here

File: cms/djangoapps/sneakpeek/middleware.py (hunk 4 above)
Comment (jlikhuva) [reply cms/djangoapps/sneakpeek/middleware.py:9]:
```source
openedx/stanford/djangoapps/sneakpeek/middleware.py
```
is a new file that implements this.

**Review by stvstnfrd — COMMENTED** [review]
posting if not already submitted...

File: common/djangoapps/util/views.py [hunk 19]
```diff
@@ -299,8 +299,6 @@ def build_error_response(status_code, field, err_msg):
     additional_info = {}
 
     required_fields = ["subject", "details"]
-    if not UserProfile.has_registered(request.user):
```
Comment (stvstnfrd) [inline common/djangoapps/util/views.py]: Should probably switch to check if not lazy

File: lms/djangoapps/branding/views.py [hunk 20]
```diff
@@ -52,14 +52,20 @@ def index(request):
     Redirects to main page -- info page if user authenticated, or marketing if not
     '''
 
-    if UserProfile.has_registered(request.user):
-        # For microsites, only redirect to dashboard if user has
-        # courses in his/her dashboard. Otherwise UX is a bit cryptic.
-        # In this case, we want to have the user stay on a course catalog
-        # page to make it easier to browse for courses (and register)
+    if request.user.is_authenticated():
+        '''
```
Comment (stvstnfrd) [inline lms/djangoapps/branding/views.py]: This comment looks like it was from edx; we should probably restore it.

File: lms/djangoapps/branding/views.py [hunk 21]
```diff
@@ -52,14 +52,20 @@ def index(request):
...
+        For microsites, onl redirect to dashboard if user has courses in their
+        dashboard. Otherwise, UX is a bit cryptic. In this case, we want the user
+        to stay on the course catalog page to make it easier to browse courses
+        (and register.)
+        '''
         if microsite.get_value(
-                'ALWAYS_REDIRECT_HOMEPAGE_TO_DASHBOARD_FOR_AUTHENTICATED_USER',
-                settings.FEATURES.get('ALWAYS_REDIRECT_HOMEPAGE_TO_DASHBOARD_FOR_AUTHENTICATED_USER', True)):
+            'ALWAYS_REDIRECT_HOMEPAGE_TO_DASHBOARD_AUTHENTICATED_USER',
```
Comment (stvstnfrd) [inline lms/djangoapps/branding/views.py]: I don't think we need a change here.

File: lms/djangoapps/courseware/access.py [hunk 22]
```diff
@@ -345,11 +347,7 @@ def can_load_forum():
         """
         Can this user access the forums in this course?
         """
-        return (
-            can_load()
-            and
-            UserProfile.has_registered(user)
-        )
+        return can_load() and user.is_authenticated()
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/access.py]: We need to explicitly check for lazy users here; lazy users should not have access to forums.

Comment (stvstnfrd) [reply lms/djangoapps/courseware/access.py]: As a potential follow-up, apply the `require_non_lazy` decorator to the forum views.

File: lms/djangoapps/courseware/access.py [hunk 23]
```diff
@@ -438,50 +436,9 @@ def check_for_staff():
...
-    "timelimit",
-    "videodev",
-    "videosequence",
-    "word_cloud",
-    "wrapper",
-]
-
-
-def _can_load_descriptor_nonregistered(descriptor):
+def _can_load_descriptor_sneakpeek(descriptor):
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/access.py]: This is sneakpeak, so we should move it to our new djangoapp.

File: lms/djangoapps/courseware/access.py [hunk 24]
```diff
@@ -438,50 +436,9 @@ def check_for_staff():
...
     return descriptor.category in NONREGISTERED_CATEGORY_WHITELIST
 
-
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/access.py]: Unrelated? Undo this?

File: lms/djangoapps/courseware/access.py [hunk 25]
```diff
@@ -587,6 +540,9 @@ def can_load():
                 or _can_access_descriptor_with_start_date(user, descriptor, course_key)
             )
         )
+        if user.is_authenticated():
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/access.py]: Oh, it looks like this moved? I think the order _may_ be important;
it shouldn't try to get a response if this check already failed.

File: lms/djangoapps/courseware/tabs.py [hunk 26]
```diff
@@ -37,7 +37,6 @@ class CoursewareTab(EnrolledTab):
     is_default = False
     is_visible_to_sneak_peek = True
 
-
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/tabs.py]: Let's not make unrelated changes.

File: lms/djangoapps/courseware/tests/test_views.py [hunk 27]
```diff
@@ -745,26 +744,6 @@ def get_text(self, course):
 
 
 @attr('shard_1')
-class TestNonRegisteredUser(TestCase):
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/tests/test_views.py:748]: Make sure we have tests for new code.

File: lms/static/sass/course/courseware/_courseware.scss [hunk 28]
```diff
@@ -663,22 +663,23 @@ section.self-assessment {
...
-  max-width: 1170px;
-  margin: 0 auto;
-  padding: 5px;
-  border: none;
-  background-color: $black;
-  color: $white;
-  line-height: 2em;
-  font-weight: bold;
-  text-align: center;
+    position: relative;
```
Comment (stvstnfrd) [inline lms/static/sass/course/courseware/_courseware.scss]: These seem unrelated; let's undo them.

File: lms/static/sass/multicourse/_course_about.scss [hunk 29]
```diff
@@ -436,7 +435,7 @@
           background: url('#{$static-path}/images/link-icon.png') left center no-repeat;
         }
       }
-      
+
```
Comment (stvstnfrd) [inline lms/static/sass/multicourse/_course_about.scss]: Revert whitespace changes.

File: lms/templates/courseware/course_about.html [hunk 30]
```diff
@@ -24,25 +26,22 @@
...
     sneakpeek_handler = function(jqXHR) {
-      if (jqXHR.status == 200) {
-        location.href = "${course_target}";
-      }
-      else {
-        $("#register_error")
-        .html("${_('An error occurred. Please try again later.')}")
-        .css("display", "block");
-      }
+        if (jqXHR.status == 200) {
```
Comment (stvstnfrd) [inline lms/templates/courseware/course_about.html]: Revert whitespace changes.

File: lms/templates/courseware/sneak_peek_register_cta.html [hunk 31]
```diff
@@ -1,26 +1,32 @@
-<%page args="user, course" />
+<%page args="user, course"/>
 <%!
-from django.conf import settings
-from django.core.urlresolvers import reverse
-from django.utils.translation import ugettext as _
+    from django.conf import settings
```
Comment (stvstnfrd) [inline lms/templates/courseware/sneak_peek_register_cta.html]: Let's not reformat this.

File: lms/templates/courseware/info.html [hunk 32]
```diff
@@ -100,7 +101,8 @@ <h2 class="page-subtitle">${course.display_name | h}</h2>
     </section>
     % endif
   </div>
-  % if sneakpeek_allowed and not regularly_registered:
-    <%include file="sneak_peek_tos_privacy.html" />
```
Comment (stvstnfrd) [inline lms/templates/courseware/info.html]: Let't not rename the file at this point.

File: requirements/edx/stanford.txt [hunk 33]
```diff
@@ -1,5 +1,5 @@
 unidecode
-xblock-image-modal==0.4.0 
+xblock-image-modal==0.4.0
```
Comment (stvstnfrd) [inline requirements/edx/stanford.txt]: Revert whitespace changes.

File: openedx/stanford/cms/envs/test.py [hunk 34]
```diff
@@ -5,8 +5,8 @@
 INSTALLED_APPS += (
     'openedx.stanford.djangoapps.register_cme',
 )
-# Remove sneakpeek during tests to prevent unwanted redirect
+
 MIDDLEWARE_CLASSES = tuple([
-    mwc for mwc in MIDDLEWARE_CLASSES
-    if mwc != 'sneakpeek.middleware.SneakPeekLogoutMiddleware'
+    c for c in MIDDLEWARE_CLASSES
```
Comment (stvstnfrd) [inline openedx/stanford/cms/envs/test.py]: Pylint usually expects variables to be at least 3 chars; let's revert the variable rename.

File: openedx/stanford/cms/envs/common.py [hunk 35]
```diff
@@ -45,11 +46,14 @@
...
+    'lazysignup',
+    'openedx.stanford.djangoapps.sneakpeek',
 )
+
 MIDDLEWARE_CLASSES += (
-    # Log out sneakpeek users
-    'sneakpeek.middleware.SneakPeekLogoutMiddleware',
+    'openedx.stanford.djangoapps.sneakpeek.middleware.BlockSneakpeekUsers',
 )
+
```
Comment (stvstnfrd) [inline openedx/stanford/cms/envs/common.py]: Revert whitespace changes.

File: common/djangoapps/student/migrations/0003_auto_20170824_1606.py [hunk 36]
```diff
@@ -0,0 +1,28 @@
...
+    dependencies = [
+        ('student', '0002_auto_20151208_1034'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name='userprofile',
+            name='nonregistered',
+        ),
+        migrations.AlterField(
```
Comment (stvstnfrd) [inline common/djangoapps/student/migrations/0003_auto_20170824_1606.py]: I think we can try to drop this;
it seems unrelated.

File: common/djangoapps/student/migrations/0003_auto_20170824_1606.py [hunk 37]
```diff
@@ -0,0 +1,28 @@
...
+            model_name='courseenrollment',
+            name='mode',
+            field=models.CharField(default='honor', max_length=100),
+        ),
+        migrations.AlterField(
```
Comment (stvstnfrd) [inline common/djangoapps/student/migrations/0003_auto_20170824_1606.py]: I think we can try to drop this;
it seems unrelated.

File: common/djangoapps/student/migrations/0003_auto_20170824_1606.py [hunk 38]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('student', '0002_auto_20151208_1034'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
```
Comment (stvstnfrd) [inline common/djangoapps/student/migrations/0003_auto_20170824_1606.py:14]: Yup, we want/expect this.

File: lms/djangoapps/courseware/migrations/0003_delete_coursepreference.py [hunk 39]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('courseware', '0002_add_coursepreference'),
+    ]
+
+    operations = [
+        migrations.DeleteModel(
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/migrations/0003_delete_coursepreference.py:14]: Yup, this seems right.

File: lms/djangoapps/lti_provider/migrations/0002_auto_20170824_1606.py [hunk 40]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (stvstnfrd) [inline lms/djangoapps/lti_provider/migrations/0002_auto_20170824_1606.py]: I think we can remove this entire file;
it looks unrelated.

File: openedx/stanford/djangoapps/sneakpeek/middleware.py [hunk 41]
```diff
@@ -0,0 +1,101 @@
...
+    'xblock_handler_noauth',
+    'xqueue_callback',
+    'about_course',
+    'course_root',
+    'info',
+ )
+LOG = logging.getLogger('edx.sneakpeek_middleware')
+
+
+class BlockSneakpeekUsers(object):
```
Comment (stvstnfrd) [inline openedx/stanford/djangoapps/sneakpeek/middleware.py]: `s/Block/Logout/`

File: openedx/stanford/cms/envs/common.py [hunk 42]
```diff
@@ -6,6 +6,7 @@
     INSTRUCTOR_QUERY_PROBLEM_TYPES,
 )
 
+AUTHENTICATION_BACKENDS += ('lazysignup.backends.LazySignupBackend',)
```
Comment (stvstnfrd) [inline openedx/stanford/cms/envs/common.py]: Maybe we don't need this on CMS since we're not allowing lazy signups there?

File: common/djangoapps/student/migrations/0003_auto_20170824_1606.py (hunk 36 above)
Comment (jlikhuva) [reply common/djangoapps/student/migrations/0003_auto_20170824_1606.py]: And, to drop this, do we just delete it from the migration file?

File: lms/urls.py [hunk 43]
```diff
@@ -1019,3 +1012,12 @@
             name='submit_financial_assistance_request'
         )
     )
+
+urlpatterns += (
```
Comment (stvstnfrd) [inline lms/urls.py]: We should move this to `openedx/stanford/djangoapps/sneakpeak/urls.py` and include it from there.

File: lms/templates/courseware/sneak_peek_register_cta.html [hunk 44]
```diff
@@ -1,26 +1,32 @@
...
-   ${_("Non-registered mode.  {tag_start}Register{tag_end} to save your course progress.")\
-     .format(platform_name=settings.PLATFORM_NAME,
-             tag_start="<a href='{}'>".format(reg_url),
-             tag_end="</a>",
-     )
-   }
-  </p>
+ % if course and is_lazy_user(user):
+    <p class="sneakpeek-warning">
+        <%
```
Comment (stvstnfrd) [inline lms/templates/courseware/sneak_peek_register_cta.html]: Can we undo the whitespace changes here too?
This seems unrelated.

File: common/djangoapps/student/views.py [hunk 45]
```diff
@@ -433,7 +432,8 @@ def signin_user(request):
...
+
+    if user.is_authenticated():
```
Comment (caesar2164) [inline common/djangoapps/student/views.py]: this should be:
```python
if request.user.is_authenticated():
```

Comment (jlikhuva) [inline common/djangoapps/student/views.py]:
```
if request.user.is_authenticated and not is_lazy_user(request.user)
```

File: common/djangoapps/student/views.py [hunk 46]
```diff
@@ -471,9 +471,8 @@ def register_user(request, extra_context=None):
     if settings.FEATURES.get('USE_CME_REGISTRATION'):
         return cme_register_user(request, extra_context=extra_context)
 
-    # Determine the URL to redirect to following login:
     redirect_to = get_next_url_for_login_page(request)
-    if UserProfile.has_registered(request.user):
+    if request.user.is_authenticated():
```
Comment (jlikhuva) [inline common/djangoapps/student/views.py]: should be same as check above.

File: common/djangoapps/student/views.py [hunk 47]
```diff
@@ -471,9 +471,8 @@ def register_user(request, extra_context=None):
     if settings.FEATURES.get('USE_CME_REGISTRATION'):
         return cme_register_user(request, extra_context=extra_context)
 
-    # Determine the URL to redirect to following login:
```
Comment (jlikhuva) [inline common/djangoapps/student/views.py]: restore comment.

File: common/djangoapps/student/views.py [hunk 48]
```diff
@@ -573,7 +572,7 @@ def is_course_blocked(request, redeemed_registration_codes, course_key):
 def dashboard(request):
     user = request.user
 
-    if not UserProfile.has_registered(user):
+    if not user.is_authenticated():
```
Comment (jlikhuva) [inline common/djangoapps/student/views.py]: and not lazy.

File: common/djangoapps/util/views.py [hunk 49]
```diff
@@ -328,7 +330,10 @@ def build_error_response(status_code, field, err_msg):
         try:
             validate_email(email)
         except ValidationError:
-            return build_error_response(400, "email", required_field_errs["email"])
+            return build_error_response(
```
Comment (jlikhuva) [inline common/djangoapps/util/views.py]: restore

File: common/lib/xmodule/xmodule/course_module.py [hunk 50]
```diff
@@ -410,6 +410,12 @@ class CourseFields(object):
         help=_("Enter true or false. If true, students can create discussion posts that are anonymous to all users."),
         scope=Scope.settings, default=True
     )
+    allow_sneak_peek = Boolean(
+        display_name=_("Allow non-registered access"),
```
Comment (jlikhuva) [inline common/lib/xmodule/xmodule/course_module.py]: non-registered ----> sneak-peek

File: lms/djangoapps/courseware/access.py [hunk 51]
```diff
@@ -345,11 +348,7 @@ def can_load_forum():
         """
         Can this user access the forums in this course?
         """
-        return (
-            can_load()
-            and
-            UserProfile.has_registered(user)
-        )
+        return can_load() and user.is_authenticated() and not is_lazy_user(user)
```
Comment (jlikhuva) [inline lms/djangoapps/courseware/access.py]: replace last 2 checks w/ new sneak_peek helper function.

File: lms/djangoapps/courseware/courses.py [hunk 52]
```diff
@@ -304,12 +305,14 @@ def get_course_info_section(request, user, course, section_key):
             context = {
                 'username': request.user.username,
                 'user_id': request.user.id,
-                'name': request.user.profile.name,
```
Comment (jlikhuva) [inline lms/djangoapps/courseware/courses.py]: Have name inside the dict instead of below.

File: lms/djangoapps/courseware/views.py [hunk 53]
```diff
@@ -729,8 +721,6 @@ def course_info(request, course_id):
             'studio_url': studio_url,
             'show_enroll_banner': show_enroll_banner,
             'url_to_enroll': url_to_enroll,
-            'sneakpeek_allowed': sneakpeek_allowed,
```
Comment (jlikhuva) [inline lms/djangoapps/courseware/views.py]: Restore `sneakpeek_allowed` and `regularly_registered (i.e is not lazy and is authenticated)`

File: lms/djangoapps/courseware/views.py [hunk 54]
```diff
@@ -1910,25 +1894,3 @@ def financial_assistance_form(request):
             }
         ],
     })
-
-
-def is_sneakpeek_allowed(user, course, course_key):
```
Comment (jlikhuva) [inline lms/djangoapps/courseware/views.py]: we still need a version of this.

File: lms/djangoapps/student_account/views.py [hunk 55]
```diff
@@ -62,7 +62,7 @@ def login_and_registration_form(request, initial_mode="login"):
     redirect_to = get_next_url_for_login_page(request)
 
     # If we're already logged in, redirect to the dashboard
-    if UserProfile.has_registered(request.user):
+    if request.user.is_authenticated():
```
Comment (jlikhuva) [inline lms/djangoapps/student_account/views.py]: exclude sneakpeek_users.
```
if request.user.is_authenticated() and not is_lazy_user(request.user)
```

File: lms/templates/courseware/course_about.html [hunk 56]
```diff
@@ -184,13 +185,13 @@
           <a href="#" class="register">
             ${_("Enroll in {course_name}").format(course_name=course.display_number_with_default) | h}
           </a>
-          % if sneakpeek_allowed and not regularly_registered:
```
Comment (jlikhuva) [inline lms/templates/courseware/course_about.html]: restore sneakpeek_allowed  and regularly_registered.

File: lms/templates/courseware/courseware.html [hunk 57]
```diff
@@ -219,12 +220,13 @@
       <div id="error-message" aria-live="polite"></div>
       <div class="courseware-results search-results" data-course-id="${course.id}" data-lang-code="${language_preference}"></div>
     </section>
+    </div>
 
-  </div>
-  % if course and user.is_authenticated() and not UserProfile.has_registered(user):
```
Comment (jlikhuva) [inline lms/templates/courseware/courseware.html]: revert whitespace changes.

File: lms/templates/help_modal.html [hunk 58]
```diff
@@ -88,7 +87,7 @@
 
     <form id="feedback_form" class="feedback_form" method="post" data-remote="true" action="/submit_feedback">
       <div id="feedback_error" class="modal-form-error" tabindex="-1"></div>
-% if not (user.is_authenticated() and UserProfile.has_registered(user)):
+% if not (user.is_authenticated()):
```
Comment (jlikhuva) [inline lms/templates/help_modal.html]:
```
if not (user.is_authenticated() and not is_lazy_user(user))
```

File: lms/templates/navigation.html [hunk 59]
```diff
@@ -63,7 +64,7 @@ <h2 class="course-header"><span class="provider">${course.display_org_with_defau
       <span class="course-name">${display_name}</span></h2>
     % endif
 
-    % if UserProfile.has_registered(user):
+    % if user.is_authenticated():
```
Comment (jlikhuva) [inline lms/templates/navigation.html]:
```
 if user.is_authenticated() and not is_lazy_user(user)
```

File: lms/templates/navigation.html [hunk 60]
```diff
@@ -118,8 +119,10 @@ <h2 class="course-header"><span class="provider">${course.display_org_with_defau
         <ul class="dropdown-menu" aria-label="More Options" role="menu">
           <%block name="navigation_dropdown_menu_links" >
             <li><a href="${reverse('dashboard')}">${_("Dashboard")}</a></li>
-            <li><a href="${reverse('learner_profile', kwargs={'username': user.username})}">${_("Profile")}</a></li>
-            <li><a href="${reverse('account_settings')}">${_("Account")}</a></li>
+            % if not is_lazy_user(user):
```
Comment (jlikhuva) [inline lms/templates/navigation.html]: remove.

File: lms/templates/courseware/courseware.html [hunk 61]
```diff
@@ -221,7 +222,7 @@
     </section>
 
   </div>
-  % if course and user.is_authenticated() and not UserProfile.has_registered(user):
```
Comment (jlikhuva) [inline lms/templates/courseware/courseware.html]: `use has_registered(user)`

File: lms/templates/courseware/sneak_peek_register_cta.html [hunk 62]
```diff
@@ -4,9 +4,9 @@
 from django.core.urlresolvers import reverse
 from django.utils.translation import ugettext as _
 
-from student.models import UserProfile
+from openedx.stanford.djangoapps.sneakpeek.utils import has_registered
 %>
-% if course and user.is_authenticated() and not UserProfile.has_registered(user):
```
Comment (jlikhuva) [inline lms/templates/courseware/sneak_peek_register_cta.html]: use `has_registered(user)`

File: common/lib/xmodule/xmodule/tabs.py [hunk 63]
```diff
@@ -383,28 +384,28 @@ def iterate_displayable(
...
-        is_user_sneakpeek=False,
     ):
         """
         Generator method for iterating through all tabs that can be displayed for the given course and
         the given user with the provided access settings.
         """
         for tab in course.tabs:
-            if (
-                tab.is_enabled(course, user=user) and
-                (not (user and tab.is_hidden)) and
```
Comment (jlikhuva) [inline common/lib/xmodule/xmodule/tabs.py]: ????

File: common/djangoapps/student/models.py [hunk 64]
```diff
@@ -1067,12 +1040,13 @@ def num_enrolled_in(cls, course_id):
...
-        enrollment_number = CourseEnrollment.objects.filter(
+        enrollment_objects = CourseEnrollment.objects.filter(
             course_id=course_id,
             is_active=1,
-            user__profile__nonregistered=0,
-        ).count()
-        return enrollment_number
+        )
+        return len(
+            [obj for obj in enrollment_objects if has_registered(obj.user)]
```
Comment (stvstnfrd) [inline common/djangoapps/student/models.py:1048]: @dcadams Is there a way to do this via the query filter?

We want all course enrollments for users who do _not_ have an entry in the `lazysignup_users` table.

File: lms/templates/courseware/sneak_peek_tos_privacy.html [hunk 65]
```diff
@@ -2,9 +2,10 @@
 from django.utils.translation import ugettext as _
 
 from student.models import UserProfile
```
Comment (stvstnfrd) [inline lms/templates/courseware/sneak_peek_tos_privacy.html]: Can we remove this import now?
Is it still being used?

File: lms/templates/help_modal.html [hunk 66]
```diff
@@ -9,6 +9,7 @@
 from xmodule.tabs import CourseTabList
 from microsite_configuration.templatetags.microsite import platform_name
 from student.models import UserProfile
```
Comment (stvstnfrd) [inline lms/templates/help_modal.html]: Can we remove this now?

**Comment by stvstnfrd** [comment]
@jlikhuva  Feel free to close this :)
